docs/en/query_language/functions/hash_functions.md

```diff
@@ -2,25 +2,72 @@
 
 Hash functions can be used for deterministic pseudo-random shuffling of elements.
 
-## halfMD5
+## halfMD5 {#hash_functions-halfmd5}
 
-Calculates the MD5 from a string. Then it takes the first 8 bytes of the hash and interprets them as UInt64 in big endian.
-Accepts a String-type argument. Returns UInt64.
-This function works fairly slowly (5 million short strings per second per processor core).
-If you don't need MD5 in particular, use the 'sipHash64' function instead.
+Produces 64-bit hash value.
 
-## MD5
+```
+halfMD5(par1, ...)
+```
+
+The function converts all the input parameters to strings, and concatenate them. Then calculates the MD5 hash value from resulting string, takes the first 8 bytes of the hash and interprets them as `UInt64` in big-endian byte order.
+
+The function works relatively slow (5 million short strings per second per processor core).
+Consider using the [sipHash64](#hash_functions-siphash64) function instead.
+
+**Parameters**
+
+The function takes a variable number of input parameters. Parameters can be any of the [supported data types](../../data_types/index.md).
+
+**Returned Value**
+
+Hash value having the [UInt64](../../data_types/int_uint.md) data type.
+
+**Example**
+
+```sql
+SELECT halfMD5(array('e','x','a'), 'mple', 10, toDateTime('2019-06-15 23:00:00')) AS halfMD5hash, toTypeName(halfMD5hash) AS type
+```
+```text
+┌────────halfMD5hash─┬─type───┐
+│ 186182704141653334 │ UInt64 │
+└────────────────────┴────────┘
+```
+
+## MD5 {#hash_functions-md5}
 
 Calculates the MD5 from a string and returns the resulting set of bytes as FixedString(16).
 If you don't need MD5 in particular, but you need a decent cryptographic 128-bit hash, use the 'sipHash128' function instead.
 If you want to get the same result as output by the md5sum utility, use lower(hex(MD5(s))).
 
-## sipHash64
+## sipHash64 {#hash_functions-siphash64}
 
-Calculates SipHash from a string.
-Accepts a String-type argument. Returns UInt64.
-SipHash is a cryptographic hash function. It works at least three times faster than MD5.
-For more information, see the link: <https://131002.net/siphash/>
+Produces 64-bit [SipHash](https://131002.net/siphash/) hash value.
+
+```
+sipHash64(par1,...)
+```
+
+This is a cryptographic hash function. It works at least three times faster than the [MD5](#hash_functions-md5) function.
+
+**Parameters**
+
+The function takes a variable number of input parameters. Parameters can be any of the [supported data types](../../data_types/index.md).
```

Comment:
it probably also converts all to string and concatenates them, but it's not stated explicitly